File: Source/JapaneseLanguageTools.Contracts/Models/Requests/SubmitFailedWordExerciseDetailsRequest.Model.cs
using System.Collections.Generic;

using JapaneseLanguageTools.Contracts.Models.Responses;

namespace JapaneseLanguageTools.Contracts.Models.Requests;

public class SubmitFailedWordExerciseDetailsRequestModel
{
    public IList<WordExerciseModel> FailedItems { get; set; } = [];

    public required GenerateWordExerciseBatchResponseModel OriginalResponse { get; set; }
}

File: Source/JapaneseLanguageTools.ReactSpa/source/models/requests/SubmitFailedWordExerciseDetailsRequest.Model.ts
import WordExerciseModel from "@/models/WordExercise.Model";
import GenerateWordExerciseBatchResponseModel from "@/models/responses/GenerateWordExerciseBatchResponse.Model";

export default interface SubmitFailedWordExerciseDetailsRequestModel {
  failedItems: WordExerciseModel[];

  originalResponse: GenerateWordExerciseBatchResponseModel;
}
